beancount-parser/src/error.rs

```diff
@@ -0,0 +1,186 @@
+                Rule::commodity_directive => "commodity directive",
+                Rule::custom_value => "a value",
+                Rule::custom_value_list => "one or more values",
+                Rule::custom => "custom directive",
+                Rule::document => "document directive",
+                Rule::event => "event directive",
+                Rule::include => "include directive",
+                Rule::note => "note directive",
+                Rule::open => "open directive",
+                Rule::option => "option directive",
+                Rule::pad => "pad directive",
+                Rule::plugin => "plugin directive",
+                Rule::price => "price directive",
+                Rule::query => "query directive",
+                Rule::transaction => "transaction directive",
+                Rule::txn_flag => "transaction flag",
+                Rule::flag_okay => "'txn' or '*'",
+                Rule::flag_warning => "'!'",
+                Rule::flag_padding => "'P'",
+                Rule::flag_summarize => "'S'",
+                Rule::flag_transfer => "'T'",
+                Rule::flag_conversions => "'C'",
+                Rule::flag_unrealized => "'U'",
+                Rule::flag_returns => "'R'",
+                Rule::flag_merging => "'M'",
+                Rule::flag_forecasted => "'#'",
+                Rule::txn_strings => "payee and narration strings",
+                Rule::posting => "posting",
+                Rule::posting_or_kv_list => "posting or metadata",
+                Rule::indented_posting_or_kv_list => "indented posting or metadata",
+                Rule::eol_posting_or_kv_list => "newline followed by indented posting or metadata",
+                Rule::price_annotation => "price annotation",
+                Rule::price_annotation_unit => "unit price annotation",
+                Rule::price_annotation_total => "total price annotation",
+                Rule::incomplete_amount => "amount and/or commodity",
+                Rule::cost_spec => "cost spec",
+                Rule::cost_spec_unit => "unit cost spec",
+                Rule::cost_spec_total => "total cost spec",
+                Rule::cost_comp_list => "a comma-separated list of cost spec components",
+                Rule::cost_comp => "a cost spec component",
+                Rule::compound_amount => "a compound amount (amount with unit and total price)",
+                Rule::file => "beancount file",
+            }
+            .to_string()
+        });
+        let location = match &err.line_col {
+            pest::error::LineColLocation::Pos(ref p) => *p,
+            pest::error::LineColLocation::Span(ref p, _) => *p,
+        };
+        ParseError {
+            kind: ParseErrorKind::InvalidInput {
+                message: format!("{}", err),
+            },
+            location,
+            source: Some(Box::new(err)),
+        }
+    }
+}
+
+pub type ParseResult<T> = Result<T, ParseError>;
```

Comment:
Not a blocking review comment, but in my experience it's nice to have type aliases towards the top since they're typically one-liners and are important in understanding the "headline" type-signature information.